File: tarefas_app/lib/models/tarefa.dart
class Tarefa {
  final int id;
  final String titulo;
  final bool concluida;

  Tarefa({
    required this.id,
    required this.titulo,
    required this.concluida,
  });

  factory Tarefa.fromJson(Map<String, dynamic> json) {
    return Tarefa(
      id: json['id'] as int,
      titulo: json['titulo'] as String,
      concluida: json['concluida'] == true || json['concluida'] == 1,
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'titulo': titulo,
    'concluida': concluida ? 1 : 0, // Converte para inteiro (0/1)
  };
}

File: tarefas-api/app/Models/Tarefa.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Tarefa extends Model
{
    use HasFactory;

    protected $fillable = ['titulo', 'concluida'];
    
    protected $casts = [
        'concluida' => 'boolean' // Converte automaticamente para boolean
    ];
    
    protected $hidden = [
        'created_at', 
        'updated_at'
    ];
}